essentials-module-utilities/src/main/java/io/github/nyliummc/essentials/mixin/utilities/ServerPlayerEntityMixin_Utilities.java

```diff
@@ -41,19 +49,43 @@ public boolean isVanished() {
 
     @Override
     public void setVanished(boolean enabled) {
-        if (true) return;
-
+        ServerPlayerEntity player = (ServerPlayerEntity) (Object) this;
         vanished = enabled;
 
-        EssentialsMod.getInstance().getServer().getPlayerManager().getPlayerList().stream().forEach((p) -> {
-            p.server.forcePlayerSampleUpdate();
-            if (p != (Object) this) {
-                if (enabled) {
-                    p.networkHandler.sendPacket(new PlayerListS2CPacket(PlayerListS2CPacket.Action.REMOVE_PLAYER, (ServerPlayerEntity) (Object) this));
-                } else {
-                    p.networkHandler.sendPacket(new PlayerListS2CPacket(PlayerListS2CPacket.Action.ADD_PLAYER, (ServerPlayerEntity) (Object) this));
+        player.setInvisible(enabled);
+
+        // Sending player ADD/REMOVE packet
+        // This updates the tablist
+        final PlayerManager playerManager = EssentialsMod.getInstance().getServer().getPlayerManager();
+        playerManager.sendToAll(
+                new PlayerListS2CPacket(
+                        enabled ? PlayerListS2CPacket.Action.REMOVE_PLAYER : PlayerListS2CPacket.Action.ADD_PLAYER,
+                        player
+                )
+        );
+
+        ThreadedAnvilChunkStorage storage = ((ServerChunkManager) player.world.getChunkManager()).threadedAnvilChunkStorage;
+        EntityTrackerAccessor_Utilities trackerEntry = ((ThreadedAnvilChunkStorageAccessor_Utilities) storage).getEntityTrackers().get(player.getEntityId());
+
+        // Starting / stopping the player tracking
+        // This actually removes the player (otherwise hacked clients still see the it with certain hacks)
+        ((EntityTrackerStreamAccessor) trackerEntry).fabric_getTrackingPlayers().forEach(
```

Comment:
Couldn't you use one of the methods from PlayerStream for this?

Reply:
Humm, haven't looked into that. I may do it a bit later.